synapse/config/server.py

```diff
@@ -302,6 +302,8 @@ def read_config(self, config: JsonDict, **kwargs: Any) -> None:
                 "public_baseurl cannot contain query parameters or a #-fragment"
             )
 
+        self.custom_well_known = config.get("custom_well_known")
```

Comment:
We should validate that the `custom_well_known` is a dictionary.

I'd suggest:
1. use an explicit fallback: `config.get("custom_well_known", {})`
2. raise a `ConfigError` if `self.custom_well_known` is not a dictionary

Reply:
Additionally, it would be great to have:
3. if `m.homeserver` is in the custom well known blob, raise a ConfigError which tells the user they should set the 
public_baseurl instead
4. Similarly if `m.identity_server` is in the custom well known blob, raise a ConfigError which tells the user they should set the default_identity_server instead